src/ui/events.js

```diff
@@ -68,6 +68,30 @@ export class MapMouseEvent extends Event {
      */
     lngLat: LngLat;
 
+    /**
+     * If a `layerId` was specified when adding the event listener with {@link Map#on}, `features` will be an array of
+     * [GeoJSON](http://geojson.org/) [Feature objects](https://tools.ietf.org/html/rfc7946#section-3.2).
+     * The array will contain all features from that layer that are rendered at the event's point.
+     * The `features` are identical to those returned by {@link Map#queryRenderedFeatures}.
+     *
+     * If no `layerId` was specified when adding the event listener, `features` will be `undefined`.
+     * You can get the features at the point with `map.queryRenderedFeatures(e.point)`.
+     *
+     * @example
+     * // logging features for a specific layer (with `e.features`)
+     * map.on('click', 'myLayerId', function(e) {
+     *     console.log('There are', e.features.length, 'features at point', e.point');
+     * });
+     *
+     * @example
+     * // logging all features for all layers (without `e.features`)
+     * map.on('click', function(e) {
+     *     var features = map.queryRenderedFeatures(e.point);
+     *     console.log('There are', features.length, 'features at point', e.point');
+     * });
```

Comment:
Nit: if we're going the ES6 way anyway, we could do it in new examples straightaway:

```suggestion
     * map.on('click', 'myLayerId', (e) => {
     *     console.log(`There are ${e.features.length} features at point ${e.point}`);
     * });
     *
     * @example
     * // logging all features for all layers (without `e.features`)
     * map.on('click', (e) => {
     *     var features = map.queryRenderedFeatures(e.point);
     *     console.log(`There are ${e.features.length} features at point ${e.point}`);
     * });
```